mealie/services/event_bus_service/event_bus_service.py

```diff
@@ -23,17 +26,32 @@ def publisher(self) -> PublisherLike:
     def get_urls(self, event_type: EventTypes) -> list[str]:
         repos = AllRepositories(self.session)
 
-        notifiers: list[GroupEventNotifierPrivate] = repos.group_event_notifier.by_group(self.group_id).multi_query(
-            {"enabled": True}, override_schema=GroupEventNotifierPrivate
-        )
+        notifiers: list[GroupEventNotifierPrivate] = repos.group_event_notifier.by_group(  # type: ignore
+            self.group_id
+        ).multi_query({"enabled": True}, override_schema=GroupEventNotifierPrivate)
 
         return [notifier.apprise_url for notifier in notifiers if getattr(notifier.options, event_type.name)]
 
-    def dispatch(self, group_id: UUID4, event_type: EventTypes, msg: str = "") -> None:
-        self.group_id = group_id
+    def dispatch(self, group_id: UUID4, event_type: EventTypes, msg: str = "", event_source: dict = None) -> None:
+        self.group_id = group_id  # type: ignore
+
+        if event_source is None:
+            event_source = {}
+
+        self.event_source = event_source
```

Comment:
I don't quite understand the need for this. Could you just use `event_source` as a function level variable and remove it from the class?

Reply:
I think I did that because of something to do with it getting passed to the `dispatch` function and then added to the async `_dispatch` function, but looking at it now I have no idea why I chose to make it a class attr as opposed to a function var ¯\_(ツ)_/¯ I'll change it